src/scheduler/scheduler.cpp

```diff
@@ -115,24 +117,12 @@ __attribute__((used)) void SysTick_Handler()
 // I don't think there's any way to make this cleaner lol
 __attribute__((used, naked)) USER_CODE void restore_regs()
 {
-    // Load fpscr first so we can avoid dirtying r0 after its popped
-    asm volatile("ldr r0, [sp, #96]\n"
-                 "vmsr fpscr, r0\n");
-
-    // Pop regs as usual
-    asm volatile("pop {r0, r1, r2, r3, r12, lr}");
-
-    // Skip SP and RETPSR. SP will be loaded last
-    // RETPSR should be ignored because we already popped when returning from exception
-    asm volatile("add sp, #8");
-
-    // Pop caller saved FP registers
-    asm volatile("vpop {s0-s15}");
-
-    // Skip FPSCR and 2 reserved regs
-    asm volatile("add sp, #12");
+    asm volatile("svc %0" ::"I"(SystemCallType::RESTORE_REGS));
+}
 
-    asm volatile("ldr pc, [sp, #-84]");
+void Scheduler::restore_current_task_regs(exception_stack_registers* regs)
+{
+    __set_PSP(reinterpret_cast<unsigned>(&regs[1]) + 4);
```

Comment:
exception_stack_regs already contains reserved regs due to alignment. I don't know why this is +4. It works though. lol